File: src/main/java/moe/zhi/intercepter/Logit.java
package moe.zhi.intercepter;

import javax.interceptor.InterceptorBinding;
import java.lang.annotation.*;

/**
 * v1.0 fix me
 *
 * @author zhimoe
 * @version 1.0
 * @since 2022/6/23
 */
@Inherited
@InterceptorBinding
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.METHOD, ElementType.TYPE})
public @interface Logit {
}

File: src/main/java/moe/zhi/interceptor/Logit.kt
package moe.zhi.interceptor

import java.lang.annotation.Inherited
import javax.interceptor.InterceptorBinding

/**
 * v1.0 fix me
 *
 * @author zhimoe
 * @version 1.0
 * @since 2022/6/23
 */
@Inherited
@InterceptorBinding
@Retention(AnnotationRetention.RUNTIME)
@Target(
    AnnotationTarget.FUNCTION,
    AnnotationTarget.CLASS
)
annotation class Logit 